src/parser/shared/modules/spells/bfa/azeritetraits/TreacherousCovenant.js

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import { formatNumber, formatPercentage } from 'common/format';
+import { calculateAzeriteEffects } from 'common/stats';
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+import TraitStatisticBox from 'interface/others/TraitStatisticBox';
+import Events from 'parser/core/Events';
+import ItemDamageTaken from 'interface/others/ItemDamageTaken';
+
+const treacherousCovenantStat = traits => Object.values(traits).reduce((obj, rank) => {
+  const stat = calculateAzeriteEffects(SPELLS.TREACHEROUS_COVENANT.id, rank);
+  obj.stat += Number(stat);
+  return obj;
+}, {
+  stat: 0,
+});
+
+const DAMAGE_MODIFIER = .15;
+
+/**
+ * Treacherous Covenant
+ * Your primary stat is increased by 151 while you are above 50% health. You take 15% increased damage while below 20% health..
+ *
+ * Example reports
+ *  Int: /report/L9AFD1kHxTrGK43t/1-Heroic+Champion+of+the+Light+-+Kill+(3:08)/20-Wiridian
+ *  Agi: /report/1YJ98MzR6qPvydGA/13-Heroic+Grong+-+Kill+(4:56)/19-Cleah
+ *  Str: /report/TMjpXkaYKVhncmfb/11-Heroic+Jadefire+Masters+-+Kill+(4:45)/11-Jhonson
+ */
+class TreacherousCovenant extends Analyzer {
+  statModifier = 0;
+  debuffActive = false;
+  extraDamageTaken = 0;
+
+  constructor(...args) {
+    super(...args);
+    this.active = this.selectedCombatant.hasTrait(SPELLS.TREACHEROUS_COVENANT.id);
+    if (!this.active) {
+      return;
+    }
+    const { stat } = treacherousCovenantStat(this.selectedCombatant.traitsBySpellId[SPELLS.TREACHEROUS_COVENANT.id]);
```

Comment:
Please add this to the statTracker too. Search for `this.statTracker.add` in existing files for an example.